launch

```diff
@@ -4,196 +4,98 @@ import argparse
-}
-DEFAULT_DEBUGGER = 'cgdb'
-assert(DEFAULT_DEBUGGER in DEBUGGERS)
-
-
-def find_triplet():
-    for triplet in ['mipsel-unknown-elf', 'mips-mti-elf']:
-        if shutil.which(triplet + '-gcc'):
-            return triplet
-
-    raise SystemExit('No cross compiler found!')
-
-
-def configure_ovpsim(args):
-    OVPSIM_OVERRIDES = {
-        'mipsle1/vectoredinterrupt': 1,
-        'mipsle1/srsctlHSS': 1,
-        'rtc/timefromhost': 1,
-        'uartCBUS/console': 1,
-        'uartCBUS/portnum': PORT,
-        'command' : '"' + ' '.join(args.args) + '"'
-    }
+from launcher import *
 
-    try:
-        os.remove('uartCBUS.log')
-        os.remove('uartTTY0.log')
-        os.remove('uartTTY1.log')
-    except OSError:
-        pass
-
-    options = [
-               '--nographics',
-               '--wallclock',
-               '--kernel', args.kernel]
-
-    if args.test or args.server:
-        OVPSIM_OVERRIDES['uartCBUS/console'] = 0
-
-    for item in OVPSIM_OVERRIDES.items():
-        options += ['--override', '%s=%s' % item]
-
-    if args.debug:
-        options += ['--port', '1234']
-
-    if args.ramdisk:
-        options += ['--ramdisk', args.ramdisk]
-    else:
-        options += ['--root', '/dev/null']
-    return options
-
-
-def configure_qemu(args):
-    options = ['-nographic',
-               '-nodefaults',
-               '-machine', 'malta',
-               '-cpu', '24Kf',
-               '-kernel', args.kernel,
-               '-append', ' '.join(args.args),
-               '-gdb', 'tcp::1234']
-
-    options += ['-serial', 'none',
-                '-serial', 'null',
-                '-serial', 'null']
-
-    if args.server:
-        options += ['-serial', 'tcp:127.0.0.1:%d,server,wait' % PORT]
-    else:
-        options += ['-serial', 'stdio']
-
-    if args.debug:
-        options += ['-S']
-
-    if args.ramdisk:
-        options += ['-initrd', args.ramdisk]
-
-    return options
-
-
-def find_simulator():
-    sim_choices = {}
-    sim_default = None
-
-    if shutil.which('qemu-system-mipsel'):
-        sim_choices['qemu'] = ['qemu-system-mipsel', configure_qemu]
-        sim_default = 'qemu'
-
-    try:
-        OVPSIM_VENDOR = 'mips.ovpworld.org'
-        OVPSIM_PLATFORM = 'MipsMalta/1.0'
-        IMPERAS_VLNV = os.environ['IMPERAS_VLNV']
-        IMPERAS_ARCH = os.environ['IMPERAS_ARCH']
-
-        ovpsim_path = os.path.join(
-            IMPERAS_VLNV, OVPSIM_VENDOR, 'platform', OVPSIM_PLATFORM,
-            'platform.%s.exe' % IMPERAS_ARCH)
-        sim_choices['ovpsim'] = [ovpsim_path, configure_ovpsim]
-        sim_default = 'ovpsim'
-    except KeyError:
-        pass
-
-    if not sim_default:
-        raise SystemExit('No simulator found!')
-
-    return sim_default, sim_choices
+if __name__ == '__main__':
+    # Check if a cross-compiler is present
+    common.find_toolchain()
 
+    # Probe which tools are available.
+    simulators, sim_default = common.find_available(SIMULATORS)
```

Comment:
That would require exposing the `simulators` module instead! I'll create a `launcher.find_simulators()` proxy in `__init__`.